### DISCUSSION TITLE
Local development without a remote build step?

### DISCUSSION THREAD

**Discussion by jschuur** [Thinkmill/keystatic#103] [Q&A] [answered]
Is it possible with the current release to have the build process run locally without it doing a git push and publishing via Vercel every time?

I understand that you can at least pick a different branch and preview your changes on a different URL, but I'd like the build process locally to work quicker and offline.

As far as I can tell, for local development Keystatic still makes commits directly to the remote repo and not the local file system. I created a branch and that didn't exist until I pulled it down locally.

What might be nice is if branches other than main, when running in local development mode, stay local first. Otherwise a change on the remote repo triggers Vercel (with the current default Vercel hosting) and this also triggers a preview build, unless you configure the '[ignore build step](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel)' option.

**Comment by simonswiss** [comment, marked as answer]
Hey!

There's actually a great feature for this scenario, via the `storage` property in the Keystatic config.

Instead of...

```js
storage: {
  kind: 'github',
  repo: {
    name: 'repo-name',
    owner: 'repo-owner',
  }
}
```

... you can do:

```js
storage: {
  kind: 'local'
}
```

~I haven't tested it out yet, but this should skip the entire `GitHub` part of things.~ I have tested it, and it works great! No GitHub App, no branches, nothing. Keystatic just creates the content files where you tell it to via the `directory` prop in your collections.

That means you bail on the whole "allow contributors to create PRs without running the project locally", but this may be exactly what you need for your specific scenario!

Hope it helps ✨

**Reply by jasikpark** [reply]
An interesting though probably more complicated idea would be to enable local git automation - to create branches and push commits entirely via a local git client, rather than through GitHub's API, though that is certainly a larger scope than using a single API
